codegen/src/main/java/RelationSchema.java

```diff
@@ -0,0 +1,64 @@
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class RelationSchema {
+    private final String name;
+    private final Map<String, Attribute> columns;
+
+    public static RelationSchema newInstance(String name, Map<String, Attribute> columns) {
+        return new RelationSchema(name, columns);
+    }
+
+    private static Map<String, Attribute> lineitemColumns;
+
+    static {
+        lineitemColumns = ImmutableMap.copyOf(new HashMap<String, Attribute>() {{
+            put("l_orderkey", new Attribute(Type.getClass("long"), 0));
+            put("l_partkey", new Attribute(Type.getClass("long"), 1));
+            put("l_suppkey", new Attribute(Type.getClass("int"), 2));
+            put("l_linenumber", new Attribute(Type.getClass("long"), 3));
+            put("l_quantity", new Attribute(Type.getClass("long"), 4));
+            put("l_extendedprice", new Attribute(Type.getClass("double"), 5));
+            put("l_discount", new Attribute(Type.getClass("double"), 6));
+            put("l_tax", new Attribute(Type.getClass("double"), 7));
+            put("l_returnflag", new Attribute(Type.getClass("char"), 8));
```

Comment:
The char type is not correct in some cases here.  For example, the varchar should be considered as String in Scala, and the l_shipinstruct and l_shipmode are fixed-length String, so they should also be String in Scala.  

In addition, please remove the l_ indicator in l_orderkey, l_partkey, and l_suppkey to adapt to the requirement of AJU-core.   The type of l_suppkey should be long.